static/app/views/settings/organizationIntegrations/configureIntegration.tsx

```diff
@@ -126,8 +140,36 @@ class ConfigureIntegration extends AsyncView<Props, State> {
     this.setState(this.getDefaultState(), this.fetchData);
   };
 
+  handleJiraMigration = async () => {
+    try {
+      const {
+        params: {orgId, integrationId},
+      } = this.props;
+
+      await this.api.requestPromise(
+        `/organizations/${orgId}/integrations/${integrationId}/issues/`,
+        {
+          method: 'PUT',
+          data: {},
+        }
+      );
+      this.setState(
+        {
+          plugins: (this.state.plugins || []).filter(({id}) => id === 'jira'),
+        },
+        () => addSuccessMessage(t('Migration in progress.'))
+      );
+    } catch (error) {
+      addErrorMessage(t('Something went wrong! Please try again.'));
+    }
+  };
   getAction = (provider: IntegrationProvider | undefined) => {
-    const {integration} = this.state;
+    const {integration, plugins} = this.state;
+    const shouldMigrateJiraPlugin =
+      provider &&
+      provider.key === 'jira' &&
```

Comment:
Will this work for Jira Server too or should it also check for the `jira_server` provider key?

Reply:
Pretty sure it's gonna say `jira_server` for the key